src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -0,0 +1,302 @@
+import com.intellij.plugins.haxe.util.HaxeClassModel;
+import com.intellij.plugins.haxe.util.HaxeMethodModel;
+import com.intellij.plugins.haxe.util.HaxePsiUtils;
+import com.intellij.plugins.haxe.util.HaxeResolveUtil;
+import com.intellij.psi.*;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.psi.util.PsiUtil;
+import com.intellij.util.IncorrectOperationException;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+public class HaxeSemanticAnnotator implements Annotator {
+
+
+  @Override
+  public void annotate(PsiElement element, AnnotationHolder holder) {
+    PsiFile file = element.getContainingFile();
+    // Analyze upon file changes
+    Long analyzedFileTime = file.getUserData(HaxeSemanticError.KEY_ANALYZED_METHOD_TIME);
+    if (analyzedFileTime == null || analyzedFileTime != file.getModificationStamp()) {
+      analyze(file);
+      file.putUserData(HaxeSemanticError.KEY_ANALYZED_METHOD_TIME, file.getModificationStamp());
+    }
+
+    List<HaxeSemanticError> errors = element.getUserData(HaxeSemanticError.KEY_HAXE_ERROR);
+    if (errors != null) {
+      for (HaxeSemanticError error : errors) {
+        Annotation annotation = holder.createErrorAnnotation(element, error.message);
+        if (error.quickfix != null) annotation.registerFix(error.quickfix);
+      }
+    }
+  }
+
+
+  static void analyze(final PsiElement element) {
+    HaxeSemanticError.clearErrors(element);
+
+    if (element instanceof PsiJavaToken) {
+      if (((PsiJavaToken)element).getTokenType() == HaxeTokenTypes.LITINT) {
+        //element.putUserData(KEY_HAXE_ERROR, "litint!");
+      }
+    }
+    else if (element instanceof HaxePackageStatement) {
+      PackageChecker.check((HaxePackageStatement)element);
+    }
+    else if (element instanceof HaxeMethod) {
+      HaxeClass clazz = HaxePsiUtils.getAncestor(element, HaxeClass.class);
+      final HaxeDeclarationAttribute overrideAttribute = HaxePsiUtils.getChild(element, HaxeDeclarationAttribute.class, "override");
+      HaxeClassModel parentClass = clazz.getModel().getExtendingClass();
+      boolean requiredOverride = false;
+
+      if (parentClass != null) {
+        HaxeMethodModel method = parentClass.getMethod(((HaxeMethod)element).getModel().getName());
+        if (method != null) {
+          requiredOverride = true;
+        }
+      }
+
+      //System.out.println(aClass);
+      if (overrideAttribute != null && !requiredOverride) {
+        HaxeSemanticError.addError(
+          element,
+          new HaxeSemanticError(
+            "Overriding nothing",
+            new HaxeSemanticIntentionAction("Fix override") {
+              @Override
+              public void run() {
+                HaxePsiUtils.replaceElementWithText(overrideAttribute, "");
```

Comment:
It just removes the override field. It can happen in the case you stop extending from another class. So just removing the override it's pretty safe, and since it is a quick fix, I think that popping the UI just for that would be too much. But yeah we can talk about that, maybe I missed something!